org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/projectimport/ProjectPreviewJob.java

```diff
@@ -77,13 +80,15 @@ protected void runToolingApiJobInWorkspace(IProgressMonitor monitor) throws Exce
     }
 
     private OmniBuildEnvironment fetchBuildEnvironment(IProgressMonitor monitor) {
-        ModelProvider modelProvider = CorePlugin.gradleWorkspaceManager().getGradleBuild(this.fixedAttributes).getModelProvider();
-        return modelProvider.fetchBuildEnvironment(FetchStrategy.FORCE_RELOAD, getToken(), monitor);
+        org.eclipse.buildship.core.workspace.GradleBuild gradleBuild = CorePlugin.gradleWorkspaceManager().getGradleBuild(this.fixedAttributes);
+        BuildEnvironment model = gradleBuild.getModelProvider().fetchModel(BuildEnvironment.class, FetchStrategy.FORCE_RELOAD, getToken(), monitor);
+        return DefaultOmniBuildEnvironment.from(model);
     }
 
     private OmniGradleBuild fetchGradleBuildStructure(IProgressMonitor monitor) {
-        ModelProvider modelProvider = CorePlugin.gradleWorkspaceManager().getGradleBuild(this.fixedAttributes).getModelProvider();
-        return modelProvider.fetchGradleBuild(FetchStrategy.FORCE_RELOAD, getToken(), monitor);
+        org.eclipse.buildship.core.workspace.GradleBuild gradleBuild = CorePlugin.gradleWorkspaceManager().getGradleBuild(this.fixedAttributes);
+        GradleBuild model = gradleBuild.getModelProvider().fetchModel(GradleBuild.class, FetchStrategy.FORCE_RELOAD, getToken(), monitor);
+        return DefaultOmniGradleBuild.from(model);
```

Comment:
Creating these temporary helper methods are fine for me.
As of moving forward with the refactoring, I suggest we dedicate time and finish it very soon. There are other feature branches that are touching the same parts of the code. Or maybe we should just merge the `remove-tooling-commons` branch to `master` after this story is complete.